### ISSUE TITLE
Rule - Resolve equals sign chain into several commands with variables in different formats

### ISSUE THREAD

**Issue by thorsten79G** [SAP/abap-cleaner#173]
Hello,

Maybe the coding is not the ideal way for the conversion, but the cleaner leads to an initial value.
The sql statement is used to read a float value. In this example, the value is a characteristic in the format date. Therefore, the float value is to be converted into a date field.
For this purpose, a second variable lv_numc8 is used because a direct conversion from float to date is not possible.  First, the float value is converted into the variable lv_numc8. In the second step, the numeric value is converted into the date variable. For convenience, this is in one line.
    
    DATA lv_numc8 TYPE n LENGTH 8.
    DATA lv_datum TYPE datum.
    
    SELECT SINGLE
        FROM ausp
        FIELDS atflv
        WHERE objek = @lv_objek
          AND atinn = @lv_atinn
          INTO @DATA(lv_float_value).
    IF sy-subrc = 0.
     lv_datum = lv_numc8 = lv_float_value.
    ENDIF.

Using the cleaner the result looks like this:

    SELECT...
    IF sy-subrc = 0.
     lv_numc8 = lv_float_value.
     lv_datum = lv_float_value.
    ENDIF.
In this particular case, where there is an assignment with different formats, the value is lost.

**Comment by jmgrassau** [comment]
Hi Thorsten, 

thanks very much for this example! That was indeed not correctly considered yet, so I guess we must change the cleanup rule "Resolve equals sign chain into several commands" to never repeat the final term, but instead to always do the assignments step by step: 

```ABAP
  " input:
  a = b = c = d = <any term>.

  " output:
  d = <any term>.
  c = d.
  b = c.
  a = b.
```

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Thorsten, 

okay, with the next release, this cleanup rule will now always make sure that all conversion steps are kept. This is also true if the rightmost term is a literal:

![image](https://github.com/SAP/abap-cleaner/assets/79639298/651f9df2-3d9d-4db3-a01c-784b677c3130)

I also added this to the description:

![image](https://github.com/SAP/abap-cleaner/assets/79639298/855607a2-5a2d-478d-bda0-38342f3492fb)

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
P.S.: This also means that the options …
![image](https://github.com/SAP/abap-cleaner/assets/79639298/308390f6-a5fc-4fbd-b523-0c5b1fac7e5f)
… were now removed.

**Comment by jmgrassau** [comment]
Hi Thorsten, 

thanks again for reporting this bug! The issue should now be fixed in version 1.9.0, which was just released.

Kind regards,
Jörg-Michael
